### ISSUE TITLE
[OUDS] Reorganize the documentation

### ISSUE THREAD

**Issue by vprothais** [Orange-OpenSource/Orange-Boosted-Bootstrap#3018]
### Prerequisites

- [x] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

The proposal is to reorganize the documentation and navigation as follow, to be more close from what is done on Design side and to help end user to find the right content that could be sometimes splitted or hard to retrieve.

We decided the following points : 
- all the component should be in a same group, and not be split between categories
- we could have a "Concept" category containing pages for each category of figma. Thoses pages would be overview and links to components of this category.
- a new "Components" entry should be added to the horizontal menu to a page listing all the components
- All customize content should be move to a new "Customize" entry in the horizontal menu with a custom lateral menu
- A new entry should be added at the end of the horizontal menu, to a page listing all useful links (DSM, icon finder, mobile documentaiton, ODS charts, a11y guidelines, ....)

Here is a ascii-art mock-up :
```
LOGO.   DOCS.   COMPONENTS.   EXAMPLES.   CUSTOMIZE     (LINKS dsm, icon finder, mobile, charts, a11y guidelines, IOT Map...)

• Getting Started (mise en place environnement, configuration)
    ○ Intro
    ○ Download
    ○ Migration/migration from boosted
    ○ …
• Concepts (Catégories de Figma)
    ○ Forms
    ○ Navigation
    ○ …
• Foundation
    ○ Approach
    ○ Tokens
    ○ Color palette
    ○ Color Mode
    ○ Reboot
• Layout
    ○ Grid
    ○ Breakpoints
    ○ …
• Components
    ○ Button
    ○ Text Input
    ○ Icons
    ○ …
• Utilities
    ○ Color
    ○ Spacing
    ○ …
• About
    ○ Team
    ○ License
```

### Motivation and context

The current documentation is organized according to what is done on bootstrap side. Even though it could be relevant for bootstrap, it is prone to difficulties for developers to find the right documentation.

The OUDS documentation should be closer to the design documentation, with same naming, at least. We should also group in the same page things that could be split in different places. The split between helpers and utilities isn't very clear for users.

Since Bootstrap is a framework aimed to be customized, it mixes documentation about usage and customization. On our side we don't want people to customize what we provide. The split between customization and use should be more clear.
